modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js

```diff
@@ -23,12 +23,13 @@ import {getFieldLocalizedValue} from '../util/fields.es';
 import {
 	getSettingsContextProperty,
 	updateField,
+	updateSettingsContextProperty,
 } from '../util/settingsContext.es';
 
-export const createDuplicatedField = (originalField, props) => {
+export const createDuplicatedField = (originalField, props, blacklist = []) => {
 	const {editingLanguageId, fieldNameGenerator} = props;
 	const label = getLabel(originalField, editingLanguageId);
-	const newFieldName = fieldNameGenerator(label);
+	const newFieldName = fieldNameGenerator(label, null, blacklist);
```

Comment:
I had to create a "blacklist" to store the fields that are being duplicated within a field group so that fields of the same `type` are not the same when duplicating the children.